gdal/port/cpl_json.cpp

```diff
@@ -0,0 +1,1051 @@
+/******************************************************************************
+ * Project:  Common Portability Library
+ * Purpose:  Function wrapper for libjson-c access.
+ * Author:   Dmitry Baryshnikov, dmitry.baryshnikov@nextgis.com
+ *
+ ******************************************************************************
+ * Copyright (c) 2016-2017 NextGIS, <info@nextgis.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#include "cpl_json.h"
+
+
+#include "cpl_error.h"
+#include "cpl_json_header.h"
+#include "cpl_vsi.h"
+
+#include "cpl_http.h"
+#include "cpl_multiproc.h"
+
+#define TO_JSONOBJ(x) static_cast<json_object*>(x)
+
+static const char *JSON_PATH_DELIMITER = "/";
+static const unsigned short JSON_NAME_MAX_SIZE = 255;
+
+//------------------------------------------------------------------------------
+// JSONDocument
+//------------------------------------------------------------------------------
+/*! @cond Doxygen_Suppress */
+CPLJSONDocument::CPLJSONDocument() : m_poRootJsonObject(nullptr)
+{
+
+}
+
+CPLJSONDocument::~CPLJSONDocument()
+{
+    if( m_poRootJsonObject )
+        json_object_put( TO_JSONOBJ(m_poRootJsonObject) );
+}
+
+CPLJSONDocument::CPLJSONDocument(const CPLJSONDocument& other)
+{
+    if( other.m_poRootJsonObject )
+    {
+        if( m_poRootJsonObject )
+            json_object_put( TO_JSONOBJ(m_poRootJsonObject) );
+        m_poRootJsonObject = json_object_get( TO_JSONOBJ(other.m_poRootJsonObject) );
+    }
```

Comment:
if other.m_poRootJsonObject == nullptr, then this->m_poRootJsonObject will be left unintiialized

Reply:
Fixed